ui/views.py

```diff
@@ -172,7 +172,9 @@ def get_vocabularies(facets):
         vocab = (vocabulary_id, vocabs[int(vocabulary_id)])
         vocabularies[vocab] = []
         for t_id, count in term_data:
```

Comment:
If it's not automatically happening in the `m2m_updated` signal then it's not happening. I just don't know if deleting it, which does remove the m2m relation, triggers that signal.